### PR TITLE
Utility tab in marine vendors+some items for it

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Adds normal binoculars, a compass, some c4 and detpacks/signallers to the marine vending machines.
Adds the compass, it just tells you your coords on the map.

## Why It's Good For The Game
Normal binoculars are hilariously hard to get, the compass adds some interesting gameplay ability to the game if your coords are ever needed.
Detpacks and C4 give normal marines some issue in using them due to the engineering skill required but they add some extra one wall destroying power to marines. And one detpack is one less magazine, so I don't see the issue.

## Changelog
:cl:
balance: added a utility tap to marine weapon vendors, has flashlights, c4, and some other utility stuff
/:cl:

### CODE REVIEW

File: code/game/objects/items/marine_gear.dm [hunk 1]
```diff
@@ -225,3 +225,16 @@
...
+
+/obj/item/compass
+	name = "compass"
+	desc = "A small compass that can tell you your coordinates on use."
+	icon_state = "compass"
+	w_class = WEIGHT_CLASS_TINY
+
+/obj/item/compass/attack_self(mob/living/user)
+	var/turf/location = get_turf(src)
+	user.show_message("<span class='warning'>After looking at the [src] you getting your bearings you can guess your general coordinates. LONGITUDE [location.x]. LATITUDE [location.y].")
```
Comment (TiviPlus) [inline code/game/objects/items/marine_gear.dm:239]: the is automatically added

File: code/game/objects/items/marine_gear.dm [hunk 2]
```diff
@@ -225,3 +225,16 @@
 	. = ..()
 	if(slot == SLOT_BELT)
 		playsound(src,'sound/machines/click.ogg', 15, FALSE, 1)
+
+//compass
```
Comment (TiviPlus) [inline code/game/objects/items/marine_gear.dm:229]: Useless comment

File: code/game/objects/items/marine_gear.dm (hunk 1 above)
Comment (TiviPlus) [reply code/game/objects/items/marine_gear.dm:239]: Missing parent call

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 3]
```diff
@@ -214,6 +214,14 @@
 		/obj/item/storage/box/visual/magazine = 30,
 		/obj/item/storage/box/visual/grenade = 10,
 		),
+		"Utility" = list(
+		/obj/item/flashlight/combat = -1,
+		/obj/item/binoculars = -1,
+		/obj/item/compass = -1,
+		/obj/item/explosive/plastique = 25,
+		/obj/item/detpack = 15,
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/marine_vending.dm:222]: No and no

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/vending/marine_vending.dm (hunk 3 above)
Comment (SplinterGP) [reply code/game/objects/machinery/vending/marine_vending.dm:222]: detpacks are fine for me, C4's are bad however, detpacks just explode like grenade or should.

**Comment by TiviPlus** [comment]
Do the detpacks too

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 4]
```diff
@@ -214,6 +214,13 @@
 		/obj/item/storage/box/visual/magazine = 30,
 		/obj/item/storage/box/visual/grenade = 10,
 		),
+		"Utility" = list(
+		/obj/item/flashlight/combat = -1,
+		/obj/item/binoculars = -1,
+		/obj/item/compass = -1,
+		/obj/item/detpack = 15,
```
Comment (Pariah919) [inline code/game/objects/machinery/vending/marine_vending.dm:221]:
```suggestion
```

**Review by Pariah919 — COMMENTED** [review]
tiiv

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**